### PR TITLE
feat(train): add validation loss with episode-level train/val split

### LINKED ISSUE #1492: Is there any plan to add a validation loss in the training pipeline, which is not dependent on simulation env.
Can we have a dataset split in the training code to run the model on a holdout validation episode to check loss on it?

### PR DESCRIPTION
**Pull request by jashshah999**
## Summary

Adds holdout validation loss to the training pipeline. This is a frequently requested feature (#1492) and is listed on the 0.6.0 roadmap under "Systematic policy verification."

**How it works:**
- `--val_split=0.1` holds out 10% of episodes for validation (deterministic split using training seed)
- `--val_freq=5000` computes validation loss every 5000 steps (defaults to eval_freq)
- Validation dataset has no image augmentation for clean evaluation
- `val/loss` is logged to wandb alongside `train/loss`

**What's added:**
- `val_split` and `val_freq` config fields in `TrainPipelineConfig`
- `compute_val_loss()` function that runs policy in eval mode on held-out episodes
- Deterministic episode split (same seed = same split)
- Separate validation DataLoader with no augmentation

**Usage:**
```bash
lerobot-train \
  --policy.path=lerobot/act_pusht \
  --dataset.repo_id=lerobot/pusht \
  --val_split=0.1 \
  --val_freq=5000
```

Disabled by default (val_split=0.0) so existing training is unaffected.

## Test plan

- [ ] Training without --val_split works as before (no regression)
- [ ] Training with --val_split=0.1 splits episodes and logs val/loss
- [ ] Val loss appears in wandb dashboard
- [ ] Split is deterministic (same seed = same episodes in train/val)

Fixes #1492.

### CODE REVIEW

**Comment by imstevenpmwork** [comment]
Hello @jashshah999 

Thanks for contributing! I'm closing this one as I believe this was implemented in: https://github.com/huggingface/lerobot/pull/3824
Feel free tore-open if this is not the case.

Kind regards!
